tensorboard/webapp/metrics/store/metrics_reducers.ts

```diff
@@ -619,6 +619,19 @@ const reducer = createReducer(
       cardStateMap: nextcardStateMap,
     };
   }),
+  on(actions.metricsCardTableExpansionToggled, (state, {cardId}) => {
+    const nextcardStateMap = {...state.cardStateMap};
+    nextcardStateMap[cardId] = {
+      ...nextcardStateMap[cardId],
+      // When card id does not exist on cardStateMap, this means expanding the table.
+      tableExpanded: !nextcardStateMap[cardId]?.tableExpanded ?? true,
```

Comment:
I don't think you need the `??`
`nextCardStateMap[undefined]?.tableExpanded`  should equal `undefined`
`!undefined` is `true`

```suggestion
      tableExpanded: !nextcardStateMap[cardId]?.tableExpanded,
```